demo/src/main/java/com/midtrans/demo/UserAddressActivity.java

```diff
@@ -196,33 +193,33 @@ private void retrieveCountryCode() {
     private void saveData() {
 
         try {
-            UserDetail userDetail = LocalDataHandler.readObject(getString(R.string.user_details), UserDetail.class);
+            CustomerDetails userDetail = MidtransSDK.getInstance().getTransactionRequest().getCustomerDetails();
 
             if (userDetail != null) {
-                Logger.i("UserAddressActivity", "userDetails:" + userDetail.getUserFullName());
+                Logger.i("UserAddressActivity", "userDetails:" + userDetail.getFirstName());
             }
 
-            ArrayList<UserAddress> userAddresses = userDetail.getUserAddresses();
+            ShippingAddress userAddresses = userDetail.getShippingAddress();
             String address = etAddress.getText().toString().trim();
             String city = etCity.getText().toString().trim();
             String zipcode = etZipCode.getText().toString().trim();
             String country = etCountry.getText().toString().trim();
 
-            if (userAddresses != null && userAddresses.get(0).getAddressType() == Constants.ADDRESS_TYPE_BOTH) {
-                UserAddress userAddress = userAddresses.get(0);
+            if (userAddresses != null) {
+                ShippingAddress userAddress = new ShippingAddress();
                 userAddress.setAddress(address);
                 userAddress.setCity(city);
-                userAddress.setZipcode(zipcode);
+                userAddress.setPostalCode(zipcode);
                 if (isCountryCodeExist(country)) {
-                    userAddress.setCountry(countryCodeSelected);
+                    userAddress.setCountryCode(countryCodeSelected);
                 } else {
                     Toast.makeText(this, "The country you choose is invalid. Please try again.", Toast.LENGTH_SHORT).show();
                     return;
                 }
+                userDetail.setShippingAddress(userAddress);
             }
 
-            userDetail.setUserAddresses(userAddresses);
-            LocalDataHandler.saveObject(getString(com.midtrans.sdk.uikit.R.string.user_details), userDetail);
+            //TODO: need to update address to customer details on transaction request
```

Comment:
will this be done before we release this SDK version?

Reply:
same here, we can remove this activity to make the user detail have static data like Snap web. Our demo app is too advance to be a demo app. how do you think...??

Reply:
see answer above

Reply:
@femmerling please re-check I was solved for the TODO